docs/data/data-grid/filtering/filtering.md

```diff
@@ -47,6 +47,80 @@ To disable `Add filter` or `Remove all` buttons, pass `disableAddFilterButton` o
 
 {{"demo": "DisableActionButtonsDataGridPro.js", "bg": "inline", "defaultCodeOpen": false}}
 
+## Header filters [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan)
+
+:::warning
+This feature is experimental, it needs to be explicitly activated using the `headerFiltering` experimental feature flag.
+
+```tsx
+<DataGridPro experimentalFeatures={{ headerFiltering: true }} {...otherProps} />
+```
+
+To use header filtering, you need to upgrade to the [Pro plan](/x/introduction/licensing/#pro-plan) or above.
+
+:::
+
+Header filtering adds a new header row using which the users can quickly add some filters to the columns. The filters added on filter panel are synchronized with the filtering row and vice versa.
+
+You can switch between different operators by clicking the operator button in the header filter cell or pressing <kbd><kbd class="key">Ctrl</kbd>+<kbd class="key">Enter</kbd></kbd> (or <kbd><kbd class="key">⌘ Command</kbd>+<kbd class="key">Enter</kbd></kbd> on macOS) when focusing on a header filter cell.
+
+{{"demo": "HeaderFilteringDataGridPro.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Clear filter icon
+
+You can render a clear icon in the header filter cell by passing `showClearIcon` to `slotProps.headerFilter`.
+
+```tsx
+<DataGridPro
+  {...data}
+  experimentalFeatures={{ headerFiltering: true }}
+  slotProps={{ headerFilter: { showClearIcon: true } }}
+/>
+```
+
+{{"demo": "HeaderFilteringClearIconDataGridPro.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Customize header filters
+
+There are multiple ways to customize header filters.
+
+#### `renderHeaderFilter` method
+
+You can use the `renderHeaderFilter` method of the `GridColDef` to customize the header filter cell for a specific column.
+
+```tsx
+const columns: GridColDef[] = [
+  {
+    field: 'isAdmin',
+    renderHeaderFilter: (params: GridColumnHeaderParams<R, V, F>) => (
+      <MyCustomHeaderFilter {...params} />
+    ),
+  },
+];
+```
+
+Following demo uses the `renderHeaderFilter` method to customize the header filter cell for the `isAdmin` column and hide the filter cell for the `phone` column.
+
+{{"demo": "CustomHeaderFilterSingleDataGridPro.js", "bg": "inline", "defaultCodeOpen": false}}
+
+#### Customize using `filterOperators`
+
+You can also update the `InputComponent` of the `filterOperator` using `GridColDef.filterOperators` to customize the header filter cell for a specific operator type.
```

Comment:
Maybe refer to https://mui.com/x/react-data-grid/filtering/#custom-input-component and remove the `disableColumnFilter` from the demo to make it clear that the same custom component is being used for both header filter and filter panel? Something like this:

```suggestion
If the filter operator has a [custom `InputComponent`](https://mui.com/x/react-data-grid/filtering/#custom-input-component), the same component is being used for the header filter.
When rendered as a header filter, the `InputComponent` also receives the `headerFilterMenu` prop that contains the filter operator menu. 
```